app/models/response.rb

```
@@ -11,9 +11,14 @@ class Response < ApplicationRecord
   alias map response_map
   delegate :reviewer_assignment, :response_assignment, :reviewee, :reviewer, to: :map
 
-  # return the questionnaire that belongs to the response
+  # return the questionnaire that belongs to the response.
+  # For varying-round assignments, used_in_round matches response.round (1 or 2).
+  # For single-round assignments, used_in_round is nil in the DB while response.round is 1,
+  # so we fall back to the first AssignmentQuestionnaire when the round-specific lookup fails.
   def questionnaire
-    reviewer_assignment.assignment_questionnaires.find_by(used_in_round: self.round).questionnaire
+    aq = reviewer_assignment.assignment_questionnaires.find_by(used_in_round: self.round)
```

RuboCop (1.87.0)

[convention] Redundant `self` detected.

(Style/RedundantSelf)